### DISCUSSION TITLE
Welcome to jpreprocess Discussions!

### DISCUSSION THREAD

**Discussion by phenylshima** [jpreprocess/jpreprocess#371] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

Note: If you have questions, please create a discussion thread from [here](https://github.com/jpreprocess/jpreprocess/discussions/new/choose).

**Comment by phudtran** [comment]
Hi, 

Would it be possible to use this crate to do g2p conversion like in pyopenjalk? I see that there's something similar in jpreprocess-pronounciation for katakana, but was wondering if it would also work for hiragana and kanji.

![image](https://github.com/user-attachments/assets/3189e10c-f6f6-447e-97cc-aca7b5ed7ffb)

**Reply by phenylshima** [reply]
@phudtran 
Thank you for your question. 
Yes, as long as you provide a dictionary which contains all the "words" in the sentence, this crate can g2p sentences with hiragana, katakana and kanji.
There is a python binding for this crate with a g2p function nearly equivalent to pyopenjtalk's, so it may help you with the implementation.
https://github.com/jpreprocess/jpreprocess/blob/py-0.1.4/bindings/python/src/binding.rs#L60-L97

(If you have more question, it may be better to create another discussion thread from [here](https://github.com/jpreprocess/jpreprocess/discussions/new/choose).)
